add targetDid to requestCredential (#55)
KILTprotocol/spec-ext-credential-api: readme.md

for email address (with one or more email address CTypes), and afterwards another requesting a credential for phone number
(with one or more phone number CTypes).

> The dApp MAY request a credential issued for a particular DID by providing it in the optional message field `owner`.
> If the message contains this field, the extension SHOULD ask the user to choose only from the credentials issued for
> this DID. Note that extensions supporting a previous version of the API might ignore this field. If the `owner` field
>  is absent, all possible credentials can be used.

Reviewer: Luke has rightfully reminded me that this message could be sent both by dApp and extension. Sorry I forgot about this earlier.
```suggestion
The sender MAY request a credential issued for a particular DID by providing it in the optional message field `owner`.
If the extension received a message with this field, it SHOULD ask the user to choose only from the credentials issued for
this DID. Note that extensions supporting a previous version of the API might ignore this field. If the `owner` field
 is absent, all possible credentials can be used.
```

for email address (with one or more email address CTypes), and afterwards another requesting a credential for phone number
(with one or more phone number CTypes).

> The message contains an optional `owner` field. The extension and the dApp SHOULD provide only credentials that are associated with the DID. If no DID is provided, all possible credentials can be used.

Reviewer 2: ```suggestion
The message contains an optional `owner` field. The extension and the dApp SHOULD provide only credentials that are associated with this DID. If no DID is provided, all possible credentials can be used.
```

Author: Adjusted: [a3c0fe9](https://github.com/KILTprotocol/spec-ext-credential-api/pull/55/commits/a3c0fe9839b2f987569d3dd5768aa6a0e4a4f130)

> The message contains an optional 'targetDid' field. The extension and the dapp MUST provide only credentials that are associated with the DID. If no DID is provided, all possible credentials can be used.

Reviewer: ```suggestion
The message contains an optional `targetDid` field. The extension and the dApp MUST provide only credentials that are associated with the DID. If no DID is provided, all possible credentials can be used.
```

> The message contains an optional `targetDid` field. The extension and the dApp SHOULD provide only credentials that are associated with the DID. If no DID is provided, all possible credentials can be used.

Reviewer: ```suggestion
The dApp MAY request a credential issued for a particular DID by providing it in the optional message field `owner`. If the message contains this field, the extension SHOULD ask the user to choose only from the credentials issued for this DID. Note that extensions supporting a previous version of the API might ignore this field. If the `owner` field is absent, all possible credentials can be used.
```

Author: adjusted: [a3c0fe9](https://github.com/KILTprotocol/spec-ext-credential-api/pull/55/commits/a3c0fe9839b2f987569d3dd5768aa6a0e4a4f130)

            requiredProperties: string[]
        }
    ]

>     /** Target DID URI*/

Reviewer: ```suggestion
    /** Target DID URI */
```

>     /** Target DID URI */

Author: Adjusted: [a3c0fe9](https://github.com/KILTprotocol/spec-ext-credential-api/pull/55/commits/a3c0fe9839b2f987569d3dd5768aa6a0e4a4f130)

>     /** Target DID URI */

Reviewer: Could you also add a newline before it? I put it there intentionally, since the code in the specification separates fields by a newline.

>     /** Target DID URI */

Author: added: [6b14288](https://github.com/KILTprotocol/spec-ext-credential-api/pull/55/commits/6b142882222e6920bc0a32d0a3d3214bfee38373)

>     targetDid: string

Reviewer 3: We called this `owner` every where else. Would it make sense to also call it owner here? It is a `RequestCredential` message. So you could argue that the owner field is the owner of the credential that is requested. `subject` would maybe be even more precise, but inconsistent with the rest of the API.

Reviewer 2: According to the spec the message can be sent both ways, but in our use cases its always the dapp.
